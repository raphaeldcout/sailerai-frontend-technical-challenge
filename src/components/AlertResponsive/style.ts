import styled from 'styled-components';

import { fadeIn } from '@/styles/keyframes';

export const Root = styled.article`
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    gap: ${({ theme }) => theme.spacings.lg};
    background-color: ${({ theme }) => theme.colors.primary};
    border-radius: ${({ theme }) => theme.borders.radius.md};
    animation: ${fadeIn} 0.4s ease-in;
    margin: ${({ theme }) => theme.spacings.lg};
    padding: ${({ theme }) => theme.spacings.lg};

    @media screen and (min-width: ${({ theme }) => theme.breakpoints.small}px) {
        display: none;
    }
`;