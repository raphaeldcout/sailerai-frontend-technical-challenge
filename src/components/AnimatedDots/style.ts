import styled, { keyframes } from 'styled-components';

const typingDots = keyframes`
  0% {
    content: '';
  }
  20% {
    content: '.';
  }
  50% {
    content: '..';
  }
  80% {
    content: '...';
  }
`;

export const AnimatedDotsStyled = styled.span`
  &::after {
    content: '';
    display: inline-block;
    animation: ${typingDots} 1.5s steps(1, end) infinite;
  }
`;