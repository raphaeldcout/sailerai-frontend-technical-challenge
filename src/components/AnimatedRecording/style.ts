import styled, { keyframes } from 'styled-components';

export const waveAnimation = keyframes`
  0%, 100% {
    transform: scaleY(1);
  }
  50% {
    transform: scaleY(1.5);
  }
`;

export const RecordingContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  width: 100px;
  height: 25px;
`;

export const Wave = styled.div`
  width: 8px;
  height: 100%;
  background-color:${({ theme }) => theme.colors.surface};
  margin: 0 2px;
  animation: ${waveAnimation} 0.6s infinite ease-in-out;
  
  &:nth-child(1) {
    animation-delay: 0s;
  }
  &:nth-child(2) {
    animation-delay: 0.1s;
  }
  &:nth-child(3) {
    animation-delay: 0.2s;
  }
  &:nth-child(4) {
    animation-delay: 0.3s;
  }
  &:nth-child(5) {
    animation-delay: 0.4s;
  }
`;