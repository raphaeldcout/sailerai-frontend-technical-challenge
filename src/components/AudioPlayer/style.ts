import styled, { css } from 'styled-components';

import { AudioPlayerContainerProps } from './types';

export const AudioPlayerContainer = styled.div<AudioPlayerContainerProps>`
  display: flex;
  flex-direction: column;
  gap: 10px;

  audio {
    width: 100%;
    outline: none;
  }

  ${({ minWidth }) => minWidth && css({ minWidth })};
`;

export const AudioPlayerStyled = styled.audio``;