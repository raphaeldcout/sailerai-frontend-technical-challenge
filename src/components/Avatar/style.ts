import ImageNext from 'next/image';
import styled, { css } from 'styled-components';

import { RootStyledProps } from './types';

export const Root = styled.figure<RootStyledProps>`
  ${({ clickable }) => clickable && css`cursor: pointer;`}
  
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  border: ${({ theme }) => theme.borders.width.thick} solid ${({ theme }) => theme.colors.border};
  border-radius: ${({ theme }) => theme.borders.radius.md};
  width: 40px;
  height: 40px;
`;

export const Image = styled(ImageNext)`
  border-radius: ${({ theme }) => theme.borders.radius.md};
`;