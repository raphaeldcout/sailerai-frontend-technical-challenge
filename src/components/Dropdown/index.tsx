import { FC } from 'react';

import {
  DropdownButtonTrigger,
  DropdownMenu,
  DropdownMenuContent
} from './style';
import { Props } from './types';

export const Dropdown: FC<Props> = ({ children, elementTrigger }) => (
  <DropdownMenu.Root>
    <DropdownMenu.Trigger asChild>
      <DropdownButtonTrigger>{elementTrigger}</DropdownButtonTrigger>
    </DropdownMenu.Trigger>

    <DropdownMenu.Portal>
      <DropdownMenuContent sideOffset={5}>{children}</DropdownMenuContent>
    </DropdownMenu.Portal>
  </DropdownMenu.Root>
);
