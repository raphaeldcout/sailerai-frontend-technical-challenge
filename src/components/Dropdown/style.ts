import styled from 'styled-components';

import { slideDownAndFade, slideLeftAndFade, slideRightAndFade, slideUpAndFade } from '@/styles/keyframes';
import * as DropdownMenu from '@radix-ui/react-dropdown-menu';

export { DropdownMenu };

export const DropdownButtonTrigger = styled.button`
  all: unset;
`;

export const DropdownMenuContent = styled(DropdownMenu.Content)`
  min-width: 220px;
  background-color: ${({ theme }) => theme.colors.primaryDark};
  border-radius: ${({ theme }) => theme.spacings.xs};
  padding: ${({ theme }) => theme.spacings.sm};
  box-shadow: 
    0px 10px 38px -10px rgba(22, 23, 24, 0.35),
    0px 10px 20px -15px rgba(22, 23, 24, 0.2);
  animation-duration: 400ms;
  animation-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
  will-change: transform, opacity;

  &[data-side='top'] {
    animation-name: ${slideDownAndFade};
  }
  &[data-side='right'] {
    animation-name: ${slideLeftAndFade};
  }
  &[data-side='bottom'] {
    animation-name: ${slideUpAndFade};
  }
  &[data-side='left'] {
    animation-name: ${slideRightAndFade};
  }
`;

export const DropdownMenuItem = styled(DropdownMenu.Item)`
  cursor: not-allowed;
  font-size: ${({ theme }) => theme.typography.fontSize.sm};
  line-height: ${({ theme }) => theme.typography.lineHeight.xs};
  color: ${({ theme }) => theme.colors.surface};
  border-radius: ${({ theme }) => theme.spacings.xs};
  display: flex;
  align-items: center;
  height: 35px;
  padding: 0 ${({ theme }) => theme.spacings.sm};
  position: relative;
  user-select: none;
  outline: none;

  &[data-highlighted] {
    background-color: ${({ theme }) => theme.colors.primary};
    color: ${({ theme }) => theme.colors.surface};
  }

  &[data-disabled] {
    color: ${({ theme }) => theme.colors.border};
    pointer-events: none;
  }
`;

export const DropdownMenuLabel = styled(DropdownMenu.Label)`
  padding: 0 ${({ theme }) => theme.spacings.sm};
  font-size: ${({ theme }) => theme.typography.fontSize.xs};
  line-height: ${({ theme }) => theme.typography.lineHeight.md};
  color: ${({ theme }) => theme.colors.border};
`;

export const DropdownMenuSeparator = styled(DropdownMenu.Separator)`
  height: 1px;
  background-color: ${({ theme }) => theme.colors.border};
  margin: ${({ theme }) => theme.spacings.sm};
`;

export const DropdownMenuArrow = styled(DropdownMenu.Arrow)`
  fill: ${({ theme }) => theme.colors.light};
`;

export const RightSlot = styled.div`
  margin-left: auto;
  padding-left: 20px;
  color: ${({ theme }) => theme.colors.surface};

  [data-highlighted] > & {
    color: ${({ theme }) => theme.colors.light};
  }

  [data-disabled] & {
    color: ${({ theme }) => theme.colors.border};
  }
`;
