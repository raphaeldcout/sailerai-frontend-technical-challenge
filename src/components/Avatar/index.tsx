import { FC } from 'react';

import { Image, Root } from './style';
import { Props } from './types';

export const Avatar: FC<Props> = ({
  alt = 'image',
  clickable = false,
  src,
  ...props
}) => (
  <Root data-testid="root-avatar" clickable={clickable} {...props}>
    <Image src={src} alt={alt} fill objectFit="cover" />
  </Root>
);
