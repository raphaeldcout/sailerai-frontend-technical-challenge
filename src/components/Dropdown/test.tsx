import '@testing-library/jest-dom';

import { Avatar } from '@/components/Avatar';
import { render } from '@/utils/jest/render';

import { Dropdown } from '.';
import { DropdownMenuItem, RightSlot } from './style';

describe('Dropdown', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <Dropdown
        elementTrigger={<Avatar src="/image-profile-1.png" clickable />}
      >
        <DropdownMenuItem>
          Meus dados <RightSlot>âŒ˜+T</RightSlot>
        </DropdownMenuItem>
      </Dropdown>
    );

    expect(container).toMatchSnapshot();
  });
});
