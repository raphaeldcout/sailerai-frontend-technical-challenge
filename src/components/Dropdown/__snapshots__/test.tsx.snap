// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match snapshot 1`] = `
<div>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    class="sc-a2edfdac-0 GxzQt"
    data-state="closed"
    id="radix-:r0:"
    type="button"
  >
    <figure
      class="sc-4d7894f7-0 jnrDtm"
      data-testid="root-avatar"
    >
      <img
        alt="image"
        class="sc-4d7894f7-1 fqwDfh"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimage-profile-1.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fimage-profile-1.png&w=640&q=75 640w, /_next/image?url=%2Fimage-profile-1.png&w=750&q=75 750w, /_next/image?url=%2Fimage-profile-1.png&w=828&q=75 828w, /_next/image?url=%2Fimage-profile-1.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-profile-1.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-profile-1.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-profile-1.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-profile-1.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </figure>
  </button>
</div>
`;
